Frontend: SettingsCluster: Add cluster name in use UI for renaming

vyncent-t opened the pull request:
This PR includes improvements to the cluster settings renaming UI, preventing users from using a name that is already taken.

### Name in use blocker UI
![image](https://github.com/user-attachments/assets/32051ca9-edd7-4e78-95e6-0728704bd20a)


- Introduced UI dialog for preventing names already in use from being used. 

### How to test the duplicate rename blocker UI
- Run headlamp in App mode
- Create a two non-dynamic clusters and let them load at home view
- Click on the settings icon in the cluster list and navigate to settings page
- Rename one of the non dynamic cluster to a name already in use by another cluster
- Notice the "Invalid custom name" UI
- Choose a name that is valid and rename it
- Navigate back to the settings page if needed and note the new custom name 
- Click on the settings tab of the renamed cluster and notice and original name displayed

illume commented:
@vyncent-t Can you please update the commit messages to remove the `.ts` extensions, and change the `src/stateless/index.ts` to `stateless: `?

vyncent-t commented:
> @vyncent-t Can you please update the commit messages to remove the `.ts` extensions, and change the `src/stateless/index.ts` to `stateless: `?

rebased and fixed it

vyncent-t commented:
> Looks like the first two messages have .ts at the end.

fixed!

illume reviewed (approved):
🎉🎈
